### ISSUE TITLE
Improve Canberra metric

### ISSUE THREAD

**Issue by sdesrozis** [pytorch/ignite#1311]
## 🚀 Feature

Actual implementation of Canberra metric does not use absolute value on terms in denominator. Moreover, `sklearn` can be used in test.

See https://arxiv.org/pdf/1411.7474.pdf 

See https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.DistanceMetric.html
